### PR TITLE
Minor improvements to the JSoupParserBolt

### PR DESCRIPTION
**Pull request by GuiForget**
- Similar to the ParserBolt, add the ability to skip the emitting of the outlinks
- Change error handling to be similar to what is done by the ParserBolt
- Minor tweaks to be in par with the ParserBolt like the name of the metrics

### CODE REVIEW

File: jsoup-parser/src/main/java/com/shopstyle/crawler/jsoup/JSoupParserBolt.java [hunk 1]
```diff
@@ -101,44 +109,12 @@ public void execute(Tuple tuple) {
...
-            CharsetMatch charsetMatch = detector.detect();
-            if (charsetMatch != null) {
-                charset = charsetMatch.getName();
-            }
-        } catch (Exception e) {
-            // ignore and leave the charset as-is
-        }
+        String charset = getContentCharset(content, metadata);
 
+        Set<String> slinks;
```
Comment (acinader) [inline jsoup-parser/src/main/java/com/shopstyle/crawler/jsoup/JSoupParserBolt.java]: why slinks?  siteLinks?

**Comment by acinader** [comment]
this looks good.  since you're in there, can you add a comment to ```link.attr("abs:href")``` to the effect:

```// abs:href tells jsoup to return fully qualified domains for relative urls. e.g.: /foo will resolve to http://shopstyle.com/foo```

File: jsoup-parser/src/main/java/com/shopstyle/crawler/jsoup/JSoupParserBolt.java [hunk 2]
```diff
@@ -156,37 +132,98 @@ public void execute(Tuple tuple) {
...
+        // with at least the URL, text content
+        declarer.declare(new Fields("url", "content", "metadata", "text"));
+        declarer.declareStream(StatusStreamName, new Fields("url", "metadata", "status"));
+    }
+
+    private String getContentCharset(byte[] content, Metadata metadata) {
+        String charset = null;
+
+        // check if the server specified a charset
+        String contentType = metadata.getFirstValue(HttpHeaders.CONTENT_TYPE);
```
Comment (acinader) [inline jsoup-parser/src/main/java/com/shopstyle/crawler/jsoup/JSoupParserBolt.java]: i like what you've done by pulling this out.  wanna rename the locals to make this more clear: ie rename contentType to contentTypeString or declaredContentType and rename ct to contentType??  not necessary...

**Comment by acinader** [comment]
looks good to me.  and helpful to me to study.  if you're inclined to do any of my suggestions cool, otherwise let me know and i'll merge.

File: jsoup-parser/src/main/java/com/shopstyle/crawler/jsoup/JSoupParserBolt.java (hunk 1 above)
Comment (GuiForget) [inline jsoup-parser/src/main/java/com/shopstyle/crawler/jsoup/JSoupParserBolt.java]: That was the name used before.

**Comment by GuiForget** [comment]
@acinader updated

**Pull request merged**